components/range_cache_memory_engine/src/engine.rs

```diff
@@ -534,6 +534,19 @@ impl RangeCacheEngine for RangeCacheMemoryEngine {
             }
         }
     }
+
+    fn region_cached(&self, region: &Region) -> bool {
+        let regions_map = self.core.region_manager().regions_map().read();
+        if let Some(meta) = regions_map.region_meta(region.get_id()) {
+            matches!(meta.get_state(), RegionState::Active | RegionState::Loading)
+        } else {
+            false
+        }
+    }
+
+    fn load_region(&self, region: &Region) {
+        let _ = self.load_region(CacheRegion::from_region(&region));
+    }
```

Comment:
How about `on_region_event(TryLoad{ ... })`?

Reply:
Why this, it is for manual load range.

Reply:
You can reuse it.